### DISCUSSION TITLE
Settings Closed Event

### DISCUSSION THREAD

**Discussion by abaljeu** [Taitava/obsidian-shellcommands#420] [Ideas - on hold]
https://forum.obsidian.md/t/global-settings-same-settings-themes-and-plugins-across-multiple-vaults/41789/134

I could automate settings sync between vaults, if whenever the Settings window closed, it would run my script.  I have no clue if it's available in Obsidian for you to detect that.

**Comment by Taitava** [comment]
Hi! I'm not aware of an Obsidian API event that would trigger when the settings modal is closed, sorry.

(It looks like there is a [`SettingTab.hide()` method in Obsidian API](https://github.com/obsidianmd/obsidian-api/blob/ea526e2459ad3f188c994862a9b106d94bf0f692/obsidian.d.ts#L3650-L3656) that (I guess) is called when the SC plugin's settings view is closed, but that's not what you are looking for, as you'd like the event to trigger also when changing other settings, not only when changing SC settings).

If anyone wants to contribute to this, the [Obsidian API](https://github.com/obsidianmd/obsidian-api/blob/master/obsidian.d.ts) could be reasearched (= try to find an event that would be triggered when Obsidian's settings modal is closed). If there's no such event (like I assume), someone could [post an API feature request to the _Developers: Plugin & API_ section in Obsidian's forums](https://forum.obsidian.md/c/developers-api/14). This discussion can be linked in the Obsidian forum post, and please also post back here a link to the forum post. Thank you! 🙂

(I'll move this discussion from _Ideas_ to _Ideas - on hold_ for now. It can be moved back if the API support gets resolved.)
